crates/biome_css_parser/src/syntax/at_rule/namespace.rs

```diff
@@ -0,0 +1,102 @@
+use crate::parser::CssParser;
+use crate::syntax::at_rule::parse_error::expected_any_namespace_url;
+use crate::syntax::value::url::{is_at_url_function, parse_url_function};
+use crate::syntax::{is_at_string, parse_regular_identifier, parse_string};
+use biome_css_syntax::CssSyntaxKind::*;
+use biome_css_syntax::{CssSyntaxKind, T};
+use biome_parser::parse_recovery::ParseRecovery;
+use biome_parser::parsed_syntax::ParsedSyntax::Present;
+use biome_parser::prelude::ParsedSyntax::Absent;
+use biome_parser::prelude::*;
+use biome_rowan::SyntaxKind;
+
+/// Checks if the current token in the parser is a `@namespace` at-rule.
+///
+/// This function verifies if the current token matches the `@namespace` rule.
+#[inline]
+pub(crate) fn is_at_namespace_at_rule(p: &mut CssParser) -> bool {
+    p.at(T![namespace])
+}
+
+/// Parses a `@namespace` at-rule in a CSS stylesheet.
+/// For specification details, see [CSS Namespaces Module](https://www.w3.org/TR/css-namespaces-3/).
+/// # Examples
+/// Basic usage in CSS:
+/// ```css
+/// @namespace url(http://www.w3.org/1999/xhtml);
+/// @namespace svg url(http://www.w3.org/2000/svg);
+/// ```
+/// This function identifies and parses these `@namespace` rules within CSS stylesheets.
+#[inline]
+pub(crate) fn parse_namespace_at_rule(p: &mut CssParser) -> ParsedSyntax {
+    if !is_at_namespace_at_rule(p) {
+        return Absent;
+    }
+
+    let m = p.start();
+
+    p.bump(T![namespace]);
+
+    if !is_at_namespace_url(p) {
+        // If we aren't at a namespace URL, then we are at a prefix,
+        // and we need to try to parse it.
+        // The problem is that both `url` and `src` are valid identifiers.
+        parse_regular_identifier(p).ok();
```

Comment:
I'm hoping to figure out how to differentiate "function tokens" from identifiers soon, which the CSS spec defines as just `<ident>(` with no spaces in between. It'll most likely be the same way that `CSS_DIMENSION_VALUE` works now, where the lexer returns a special kind that can be converted to the actual kind (in this case, `IDENT`).

I'd thought about making the parser not skip over whitespace, but i think that just makes all of the steps in the parser more complicated, while re-casting a token to a different kind is effectively free and makes checks incredibly fast, too.

Reply:
I agree, having `function_tokens` would be better.

Reply:
I can go ahead and get started on that. I've got a good idea in mind already.